### DISCUSSION TITLE
Best Practice Design for "Offline First" With Realm Sync (.net)? (Realm, RealmSyncApp, Configuration)

### DISCUSSION THREAD

**Discussion by dgerding** [realm/realm-dotnet#2672] [Q&A] [answered]
Hi,
I'm in the process of trying to convert to the Sync version of Realms, but don't quite understand the right approach if I really want an "offline", "sync when able" or "sync manually" assumption?

Here's what I'm after...
I create an "email+pw Realm User" from app with, requisite, Internet access on "first app use".  After that, assume the app starts up _without_ Internet. Or that it is assumed unavailable or intermittent.

Without Internet, for a **Realm Sync** app:
- How do I create get a legit Realm reference during app startup with NO internet? (Do I use a cached copy of the Realm UserID for the Partition value?)
- Should I create a new Realm reference once app has Internet access? Do I "have to"
- Is there a way to manually invoke sync instead of automagically?
- And what about EncryptionKey for client-side at rest encryption?

Thanks,
Dave Gerding

**Comment by papafe** [comment, marked as answer]
Hi @dgerding

When using  MongoDB Realm Sync you don't really need to worry about the connection status, as realm will work in the same way both when online and offline. When online realm will sync all the necessary changes in the background, but you don't need to do anything specific to make it work.

Regarding more specifically your questions:
- You're correct that at least the first time the user need to have an internet connection and do a login. After the first login, you can open the realm using the user from [App.CurrentUser](https://docs.mongodb.com/realm-sdks/dotnet/latest/reference/Realms.Sync.App.html#Realms_Sync_App_CurrentUser). 
- You don't need to create a new Realm reference once the user has internet access. All the synchronisation happens automatically.
- You can find methods to pause/resume the sync session [here](https://docs.mongodb.com/realm/sdk/dotnet/examples/sync-changes-between-devices/#std-label-dotnet-sync-data). Is there a specific reason why you would like to do so? 
- You can still use encryption with synced realm but please note that MongoDB Realm only encrypts the data on the device, and stores the data unencrypted in your MongoDB Atlas data source. The transfer between client server is fully encrypted.

I hope that this answer your doubts.

**Comment by dgerding** [comment]
Hi, thanks for the reply. I'd missed this response and am delighted about the Sync stop mechanism. I'll go and remove the related Feature Request I added in the interim.

Great news - doubts answered :) 

(Fyi, one of the tech support staff told me there was no way at all to regulate sync... I'll send them this and thanks)
